msu/ui/tooltip/classes/abstract_tooltip.nut

```diff
@@ -0,0 +1,29 @@
+::MSU.Class.Tooltip <- class
+{
+	Title = null;
+	Text = null;
+
+	constructor(_title, _text)
+	{
+		::MSU.requireOneFromTypes(["string", "function"], _title, _text);
+		this.Title = _title;
+		this.Text = _text;
+	}
```

Comment:
what about converting title and text to a lambda function if it's a string during initialization so you don't have to worry about this?
```squirrel
this.Title = typeof _title == "string" ? @() _title : _title;
this.Text = typeof _text == "string" ? @() _text : _text;
```
That way you can always assume it's a function. 'Parse, don't validate' xD

Reply:
Has been implemented with getters and setters.